### DISCUSSION TITLE
Issues packaging Spacy 3.0.6 (en_core_web_sm.load())

### DISCUSSION THREAD

**Discussion by JimMcDonough** [pyinstaller/pyinstaller#5896] [Help] [answered]
I have an application that uses spacy and was able to resolve this issue with an older version of spacy from help I found online.  The hook files and hidden imports from old posts do not solve work for this version.  In particular my exe is not able to run en_core_web_sm.load() after importing spacy and en_core_web_sm.  I am thinking this has to do with spacy_legacy architectures and not being found and resolved.  I could be wrong.  I seeing if anyone has gotten spacy 3.0.6 to work in an exe in the last few months.  

 File "en_core_web_sm\__init__.py", line 11, in load
  File "spacy\util.py", line 516, in load_model_from_init_py
  File "spacy\util.py", line 391, in load_model_from_path
  File "spacy\util.py", line 428, in load_model_from_config
  File "spacy\language.py", line 1667, in from_config
  File "spacy\language.py", line 773, in add_pipe
  File "spacy\language.py", line 659, in create_pipe
  File "thinc\config.py", line 731, in resolve
  File "thinc\config.py", line 781, in _make
  File "thinc\config.py", line 837, in _fill
  File "thinc\config.py", line 836, in _fill
  File "thinc\config.py", line 1027, in make_promise_schema
  File "spacy\util.py", line 139, in get
catalogue.RegistryError: [E893] Could not find function 'spacy.Tok2Vec.v1' in function registry 'architectures'. If you're using a custom function, make sure the code is available. If the function is provided by a third-party package, e.g. spacy-transformers, make sure the package is installed in your environment.

Available names: spacy.CharacterEmbed.v2, spacy.EntityLinker.v1, spacy.HashEmbedCNN.v2, spacy.MaxoutWindowEncoder.v2, spacy.MishWindowEncoder.v2, spacy.MultiHashEmbed.v2, spacy.PretrainCharacters.v1, spacy.PretrainVectors.v1, spacy.Tagger.v1, spacy.TextCatBOW.v1, spacy.TextCatCNN.v1, spacy.TextCatEnsemble.v2, spacy.TextCatLowData.v1, spacy.Tok2Vec.v2, spacy.Tok2VecListener.v1, spacy.TorchBiLSTMEncoder.v1, spacy.TransitionBasedParser.v1, spacy.TransitionBasedParser.v2
[16532] Failed to execute script DDP_GUI_JM_Drop_Down_Prototype_11_16_20

**Comment by rokm** [comment, marked as answer]
Have you tried collecting submodules and metadata for `spacy_legacy`?

For the reference, the following program

```python
# program.py
import spacy
from spacy.lang.en.examples import sentences

nlp = spacy.load("en_core_web_sm")
doc = nlp(sentences[0])
print(doc.text)
for token in doc:
    print(token.text, token.pos_, token.dep_)
```

requires the following PyInstaller (4.3) command-line on linux:

```
pyinstaller --clean --noconfirm program.py \
--hidden-import srsly.msgpack.util \
--hidden-import cymem \
--hidden-import cymem.cymem \
--hidden-import preshed.maps \
--collect-submodules thinc \
--collect-data thinc \
--collect-submodules blis \
--collect-submodules spacy \
--collect-data spacy \
--collect-submodules en_core_web_sm \
--collect-data en_core_web_sm \
--copy-metadata en_core_web_sm \
--collect-submodules spacy_legacy \
--copy-metadata spacy_legacy
```

where adding the last two switches fixed the missing `spacy.Tok2Vec.v1` error.

**Reply by JimMcDonough** [reply]
I had this 
datas.extend(PyInstaller.utils.hooks.collect_data_files('spacy_legacy',include_py_files = True))

but I will try it like you have it and see what happens.  Thanks you.

**Reply by JimMcDonough** [reply]
This worked thanks.

**Reply by DarkSoliditi** [reply]
Hey I'm also have trouble packaging spacy with pyinstaller however I'm not getting any error messages. It just fails on the line where I set nlp = spacy.load("en_core_web_trf"). Do you think adding the hidden imports would work in my case?
